### PR TITLE
Media type differentiation for MAL Id retrieval

### PR DESCRIPTION
**Pull request by DeadlyEmbrace**
In some cases a Mal Id can link to multiple media entries on AniList
(See [Neverland S2](https://myanimelist.net/anime/39617). It's Id for Media type `Anime` links to [Neverland S2](https://anilist.co/edit/anime/108725) and for Media type `Manga` to [Vermillion](https://anilist.co/edit/manga/69617)

This request adds a method to the Client that allows the user to specify the media type when requesting an entry by Mal Id

### CODE REVIEW

File: Anilist4Net.Test/MediaTests.cs [hunk 1]
```diff
@@ -95,5 +95,15 @@ public async Task HigurashiTest()
 			IsNotNull(media.StartDate.ToDate());
 			AreEqual(new DateTime(2009, 1, 1, 0, 0, 0, DateTimeKind.Unspecified), media.StartDate.ToDate());
 		}
+
+		// Caters for cases where there are more than one Mal Id that is the same (for example an anime and a manga) by allowing the user to specify the media type they expect
+        [TestCase(MediaTypes.ANIME, "Yakusoku no Neverland 2")]
+        [TestCase(MediaTypes.MANGA, "Vermillion")]
+		public async Task SettingExpectedMediaTypeCorrectlyDifferentiateMatchingMalIds(MediaTypes typeToRetrieve, string expectedTitle)
```
Comment (yellowsink) [inline Anilist4Net.Test/MediaTests.cs]:
```suggestion
		public async Task MalIdByTypeTest(MediaTypes typeToRetrieve, string expectedTitle)
```
The test name here is a little long winded, but apart from that this PR looks good :)

**Review by yellowsink — APPROVED** [review]

**Comment by yellowsink** [comment]
Cool thanks! I didn't realise MAL IDs worked this way lol (oops!)

**Comment by DeadlyEmbrace** [comment]
👍 Neither did I, I always though Mal used a single block of Ids like AniList... seems not.
Was quite surprised when some of my shows were replaced with manga.

**Comment by yellowsink** [comment]
Yeah. Seems weird to not share a pool of IDs anilist-style. But oh well that's MAL's choice :shrug:

**Pull request merged**